### PR TITLE
Added Submariner Install

### LINKED ISSUE #21: Support flat network between multiclusters
**What would you like to be added**:

There are multi solutions to make the connectivity between different clusters.
Currently we can make use of istio to do that.

There are many other light weight solutions like [submariner](https://submariner.io/) and [NSM](https://networkservicemesh.io/), both have their use cases.

I think we should make use of them to achieve flat network between clusters.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #21

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/plugin/submariner/install.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"kurator.dev/kurator/pkg/util"
+)
+
+func (p *SubmarinerPlugin) runInstall() error {
+	// 1.generate kubeconfig for each cluster
```
Comment (zirain) [inline pkg/plugin/submariner/install.go:34]: for each member cluster?

File: pkg/plugin/submariner/install.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	memClusters, err := p.generateKubeConfiguration()
+	if err != nil {
+		return err
+	}
+	if len(memClusters) < 2 {
+		return fmt.Errorf("memcluster number %d < 2", len(memClusters))
+	}
+	// 2. install broker in the first memcluster
```
Comment (zirain) [inline pkg/plugin/submariner/install.go:42]: `first` sort by cluster name?

File: pkg/plugin/submariner/install.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	err = p.installBroker(memClusters[0])
+	if err != nil {
+		return fmt.Errorf("install broker failed: %v", err)
+	}
+	// 3. connect all clusters to the Broker
+	err = p.connectBroker(memClusters)
```
Comment (zirain) [inline pkg/plugin/submariner/install.go:48]: the cluster which broker installed in also need connect to broker?

File: pkg/plugin/submariner/plugin.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		// For other tar format, we use `tar` command directly
+		subctl := fmt.Sprintf("subctl-%s-%s-%s", istioComponent.Version, util.OSExt(), runtime.GOARCH)
+		pkg := subctl + ".tar.xz"
+		url, _ := util.JoinUrlPath(istioComponent.ReleaseURLPrefix, istioComponent.Version, pkg)
+		fileName := path.Join(p.installPath, pkg)
+		// download subctl tarball
+		if _, err := util.DownloadResource(url, fileName); err != nil {
+			return "", fmt.Errorf("unable to get subctl binary %q: %w", p.installPath, err)
+		}
+		cmd := exec.Command("tar", "-xvf", fileName, "--strip-components", "1", "-C", p.installPath)
```
Comment (zirain) [inline pkg/plugin/submariner/plugin.go:114]: should we do this in `util.DownloadResource`?

File: pkg/plugin/submariner/install.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/plugin/submariner/install.go:42]: By the order of joining

File: pkg/plugin/submariner/plugin.go (hunk 4 above)
Comment (hzxuzhonghu) [reply pkg/plugin/submariner/plugin.go:114]: Plan to refactor the function to process any format

File: pkg/plugin/submariner/install.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pkg/plugin/submariner/install.go:48]: Seems need to

**Comment by hzxuzhonghu** [comment]
>by default, a limitation of Submariner is that it doesn’t handle overlapping CIDRs (ServiceCIDR and ClusterCIDR) across clusters. If the CIDRs of two or more clusters overlapping, How do we handle it, quit the installation or install the Global Private Network component of Submariner?

Good question, the solution can only handle none overlapping scenario. We can handle this later

**Pull request merged**